src/patterns/nav-a-z/NavAZ.tsx

```diff
@@ -69,18 +76,21 @@ const LinkItem: React.FC<AsElementLink<HTMLAnchorElement>> = ({
   asElement: Component = 'a',
```

Comment:
Not for now, but we can probably do some magic here with a bit of rework on the AsElementLink so that we can pass in for example a react-router/any other lib router Link element and infer its props to LinkItem

Reply:
I was thinking about this - as it'll have the props of a HTMLAnchorElement regardless of the component actually being used. It's definitely some Typescript inference magic to be done more generally.